homeassistant/helpers/config_entry_flow.py

```diff
@@ -104,15 +105,29 @@ def register_discovery_flow(
     domain: str,
     title: str,
     discovery_function: DiscoveryFunctionType,
-    connection_class: str,
+    connection_class: str | UndefinedType = UNDEFINED,
 ) -> None:
     """Register flow for discovered integrations that not require auth."""
+    if connection_class is not UNDEFINED:
+        _LOGGER.warning(
+            (
+                "The %s (%s) integration is setting a connection_class"
+                " when calling the 'register_discovery_flow()' method in its"
+                " config flow. The connection class has been deprecated and will"
+                " be removed in a future release of Home Assistant."
+                " If '%s' is a custom integration, please contact the author"
+                " of that integration about this warning.",
+            ),
+            title,
+            domain,
+            title,
```

Comment:
```suggestion
            domain,
```
I think domain makes more sense here, title is the name of the config entry, not the integration